src/calibre/gui2/main.py

```diff
@@ -400,11 +396,11 @@ def run_gui(opts, args, listener, app, gui_debug=None):
         if getattr(runner.main, 'debug_on_restart', False) or gui_debug is not None:
             run_in_debug_mode()
         else:
-            import subprocess
+            import shlex, subprocess
             if hasattr(sys, 'frameworks_dir'):
                 app = os.path.dirname(os.path.dirname(os.path.realpath(sys.frameworks_dir)))
                 prints('Restarting with:', app)
-                subprocess.Popen('sleep 3s; open ' + shellquote(app), shell=True)
+                subprocess.Popen('sleep 3s; open ' + shlex.quote(app), shell=True)
```

Comment:
shlex.quote is not available in python 2. And the sleep is done via the shell not time.sleep() so that it does not block the main thread. Although, since this is macOS specific, one could use preexec_fn instead.

Reply:
On the other hand, given the warnings about preexec_fn deadlocking, better to just leave it to use the shell

Reply:
Commit dropped...

(using run_calibre_debug seems like a fine solution)